### PR TITLE
Python API revisions

### PR DESCRIPTION
**Pull request by zackchase**
Second pass over all .py docstrings, covered module subdirectory this time, and moved all docstrings into closer compliance with NumPy doc standard.

### CODE REVIEW

File: python/mxnet/name.py [hunk 1]
```diff
@@ -16,25 +16,25 @@ def __init__(self):
...
-        This is default implementation.
-        When user specified a name,
-        the user specified name will be used.
+        This is the default implementation.
+        If the user specifies a name,
+        the user-specified name will be used.
 
-        When user did not, we will automatically generate a
-        name based on hint string.
+        When user does not speciry a name, we automatically generate a
```
Comment (sergeykolychev) [inline python/mxnet/name.py]: small typo, speciRy

**Comment by zackchase** [comment]
@sergeykolychev fixed. thanks for the eyes.

**Comment by madjam** [comment]
This can be merged once the pylint "line too long" errors are fixed.

**Comment by zackchase** [comment]
Fixed. Thanks Madan.

**Comment by mli** [comment]
@pluskid this pr changed binded to bound for `BaseModule`

but i suggest to split this pr into two: 1) changing binded. 2) doc fix, to make review and later issue locating easier.

**Comment by pluskid** [comment]
Thanks a lot!
Could we keep the interface `binded` for a while with a deprecate warning? Something like this:

```python
@property
def binded(self):
    from warnings import warn
    warn("binded is deprecated, use bound instead", DeprecationWarning)
    return self.bound
```

**Comment by zackchase** [comment]
Hi Chiyuan,

We did precisely this, see:

    @property
    def binded(self):
        warnings.warn(
            'binded has been deprecated. ' + \
            'Please use bound instead.',
            DeprecationWarning, stacklevel=2)
        return self.bound


However although it appears to work locally for me I'm having some issue
when I run unit tests locally. This might owe to me having an outdated
version of the build locally. Can someone else please help verify that this
change does not break anything before merging?

**Comment by piiswrong** [comment]
do we really need to change binded to bound? It can potentially cause a lot of troubles.

**Comment by zackchase** [comment]
Binded appears to not be a real word. The correct past tense is  "bound".
See for reference in computer science any of Linux reference material on
sockets. I suggest we use reasonable, intelligible terms wherever possible
and better to make the change now than further down the road. Why should
things break as long as we keep

@property binded

for backward compatibility?

**Comment by madjam** [comment]
I agree that this looks benign. However, code changes will require additional scrutiny and will potentially hold up rest of the doc related changes that can be merged.
My recommendation is to add  `TODO` to fix issues like this and move forward with just the doc changes.

**Comment by zackchase** [comment]
I think this one we should put in that effort now and just make it right
and merge in. For the future you are right. However this will allow me to
move on quickly and focus on .io

**Comment by piiswrong** [comment]
This could lead to all kinds of problems. At the least people using custom modules will see a lot of deprecation message.

Also since bind lacks a setter, custom modules bind method will break.

I think we should revert the code change and merge in the docs.

**Comment by zackchase** [comment]
Ok ill revert

**Comment by zackchase** [comment]
Generally we need to make sure variable names are reasonable (and correctly
spelled) when reviewing commits.

**Comment by eric-haibin-lin** [comment]
@zackchase Great effort for the API revisions. Thanks. What's the plan for newly added APIs for future PRs - do we have some protocol? We need to make sure the new ones are reasonable, too

**Comment by zackchase** [comment]
Hi - I changed bound back to binded so that code could be merged in without breaking the universe. Can someone please take a look and make sure everything checks out and then merge in?

**Comment by piiswrong** [comment]
looks good. waiting for tests to pass

**Pull request merged**